pkgs/applications/editors/kakoune/default.nix

```diff
@@ -2,34 +2,61 @@
 
 with stdenv.lib;
 
-stdenv.mkDerivation rec {
-  name = "kakoune-${version}";
-  version = "2019.01.20";
-  src = fetchFromGitHub {
-    repo = "kakoune";
-    owner = "mawww";
-    rev = "v${version}";
-    sha256 = "04ak1jm7b1i03sx10z3fxw08rn692y2fj482jn5kpzfzj91b2ila";
-  };
-  nativeBuildInputs = [ pkgconfig ];
-  buildInputs = [ ncurses asciidoc docbook_xsl libxslt ];
-  makeFlags = [ "debug=no" ];
-
-  postPatch = ''
-    export PREFIX=$out
-    cd src
-    sed -ie 's#--no-xmllint#--no-xmllint --xsltproc-opts="--nonet"#g' Makefile
-  '';
-
-  preConfigure = ''
-    export version="v${version}"
-  '';
-
-  meta = {
-    homepage = http://kakoune.org/;
-    description = "A vim inspired text editor";
-    license = licenses.publicDomain;
-    maintainers = with maintainers; [ vrthra ];
-    platforms = platforms.unix;
+let
+  withPlugins = plugins: package.overrideAttrs (super: {
+    buildInputs = plugins ++ super.buildInputs;
+
+    postInstall = foldr (input: script: script + ''
+      ( cd "${input}"
+        printf 'Installing autoload scripts from ${input}... '
+        if [[ -d share/kak/autoload ]]; then
+          printf '(found)\n'
+          find share/kak/autoload -type d -exec mkdir -p "$out/{}" \;
+          find share/kak/autoload -type f -exec ln -sfv "${input}/{}" "$out/{}" \;
+        else
+          printf '(none found)\n'
+        fi )
+    '') "" plugins;
+
+    passthru = super.passthru // {
+      withPlugins = morePlugins: withPlugins (morePlugins ++ plugins);
+    };
+  });
+
+  package = stdenv.mkDerivation rec {
+    name = "kakoune-${version}";
```

Comment:
you can use `pname = "kakoune";` instead

Reply:
done